### DISCUSSION TITLE
Show Diagnostic Messages Inline

### DISCUSSION THREAD

**Discussion by ghost** [VonHeikemen/lsp-zero.nvim#97] [Q&A] [answered]
Hello! 👋

By default, I see column signs for issues and can view diagnostic messages in a floating window via the `gl` map. This seems to be the expected behavior as part of the `recommended` preset.

How can change the behavior to instead show the diagnostic messages inline? I've tried setting `float = false` in options passed when calling `set_preferences`, but that doesn't seem to work.

Any help would be greatly appreciated.
Thanks.

**Comment by VonHeikemen** [comment, marked as answer]
I think you are looking for "virtual text". You can enable it using `vim.diagnostic.config`.

```lua
local lsp = require('lsp-zero')
lsp.preset('recommended')

lsp.setup()

vim.diagnostic.config({
  virtual_text = true,
})
```

**Reply by ghost** [reply]
That's exactly what I was looking for. Thank you!
